clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -743,8 +750,11 @@ public void handleResponse(AbstractResponse response) {
                     lookupCoordinator(FindCoordinatorRequest.CoordinatorType.TRANSACTION, transactionalId);
                     reenqueue();
                     return;
-                } else if (error == Errors.COORDINATOR_LOAD_IN_PROGRESS || error == Errors.CONCURRENT_TRANSACTIONS
-                        || error == Errors.UNKNOWN_TOPIC_OR_PARTITION) {
+                } else if (error == Errors.CONCURRENT_TRANSACTIONS) {
+                    retryBackoffMs = 5;
```

Comment:
I updated the code to only lower the retryBackoff on the first add partitions. Further, it lowers the minimum off the configured retryBackoff and the one returned by the handler. 

Finally, I set the new retryBackoff to 15ms..